### PR TITLE
Revert updating Django 3 to Django 4

### PR DESCRIPTION
**Pull request by SzymJ**
I want to merge this change because it downgrades Django version.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by SzymJ** [comment]
`psycopg2` and `pyjwt` need to be locked at this moment and handled in second PR.

File: requirements.txt [hunk 1]
```diff
@@ -1,170 +1,165 @@
...
-vine==5.0.0 ; python_version >= "3.9" and python_version < "3.10"
-watchfiles==0.17.0 ; python_version >= "3.9" and python_version < "3.10"
-wcwidth==0.2.5 ; python_version >= "3.9" and python_version < "3.10"
-weasyprint==56.1 ; python_version >= "3.9" and python_version < "3.10"
-webencodings==0.5.1 ; python_version >= "3.9" and python_version < "3.10"
-websockets==10.3 ; python_version >= "3.9" and python_version < "3.10"
-wrapt==1.14.1 ; python_version >= "3.9" and python_version < "3.10"
-zipp==3.9.0 ; python_version >= "3.9" and python_version < "3.10"
-zopfli==0.2.1 ; python_version >= "3.9" and python_version < "3.10"
+adyen==4.0.0
```
Comment (NyanKiyoshi) [inline requirements.txt:1]: Can you use Poetry≥1.2.0? Looks like it was generating via Poetry <1.2.0 thus the diffs are way too big

File: pyproject.toml [hunk 2]
```diff
@@ -51,7 +51,7 @@ documentation = "https://docs.saleor.io/"
   jaeger-client = "^4.5.0"
   openpyxl = "^3.0.3"
   django-cache-url = "^3.1.2"
-  pyjwt = ">=2.4.0,<3.0.0"
+  pyjwt = "2.5.0"
```
Comment (NyanKiyoshi) [inline pyproject.toml:54]: Add a comment why it's locked + GitHub issue link

File: requirements.txt (hunk 1 above)
Comment (SzymJ) [reply requirements.txt]: true, I updated Poetry and generated files one more time.

File: pyproject.toml [hunk 3]
```diff
@@ -39,7 +39,7 @@ documentation = "https://docs.saleor.io/"
   opentracing = "^2.3.0"
   phonenumberslite = "^8.12.25"
   prices = "^1.0"
-  psycopg2 = "^2.8.3"
+  psycopg2 = "2.9.4"
```
[the remark this answers is not in the document]
Comment (SzymJ) [reply pyproject.toml]: I was able to unlock this one.

File: pyproject.toml (hunk 2 above)
Comment (SzymJ) [reply pyproject.toml:54]: It is commented, I will create a issue for that

**Review by maarcingebala — APPROVED** [review]

File: pyproject.toml [hunk 4]
```diff
@@ -51,7 +51,7 @@ documentation = "https://docs.saleor.io/"
   jaeger-client = "^4.5.0"
   openpyxl = "^3.0.3"
   django-cache-url = "^3.1.2"
-  pyjwt = ">=2.4.0,<3.0.0"
+  pyjwt = "2.5.0" # locked because version 2.6.0 changed iat validation that breaks Saleor tests, will be handled in seperate PR
```
Comment (NyanKiyoshi) [inline pyproject.toml:54]: A link to the issue to make it easier to find it
```suggestion
  pyjwt = "2.5.0" # Version 2.6.0 changed iat validation which causes tests to fail: https://github.com/saleor/saleor/issues/11047
```

**Review by NyanKiyoshi — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Pull request merged**